runtime/foucoco/src/lib.rs

```diff
@@ -973,6 +995,80 @@ impl orml_tokens_management_extension::Config for Runtime {
 	type AssetDeposit = AssetDeposit;
 }
 
+pub struct AllowedCurrencyIdVerifierImpl;
+impl treasury_buyout_extension::AllowedCurrencyIdVerifier<CurrencyId>
+	for AllowedCurrencyIdVerifierImpl
+{
+	fn is_allowed_currency_id(currency_id: &CurrencyId) -> bool {
+		matches!(
+			currency_id,
+			CurrencyId::XCM(0) | CurrencyId::XCM(1) | CurrencyId::XCM(2) | CurrencyId::XCM(6)
+		)
+	}
+}
+pub struct OracleWrapper(Oracle);
+impl treasury_buyout_extension::PriceGetter<CurrencyId> for OracleWrapper {
+	#[cfg(not(feature = "runtime-benchmarks"))]
+	fn get_price<FixedNumber>(currency_id: CurrencyId) -> Result<FixedNumber, DispatchError>
+	where
+		FixedNumber: FixedPointNumber + One + Zero + Debug + TryFrom<FixedU128>,
+	{
+		let key = OracleKey::ExchangeRate(currency_id);
+		let asset_price = Oracle::get_price(key.clone())?;
+
+		let converted_asset_price = FixedNumber::try_from(asset_price);
+
+		match converted_asset_price {
+			Ok(price) => Ok(price),
+			Err(_) => Err(DispatchError::Other("Failed to convert price")),
+		}
+	}
+	#[cfg(feature = "runtime-benchmarks")]
+	fn get_price<FixedNumber>(currency_id: CurrencyId) -> Result<FixedNumber, DispatchError>
+	where
+		FixedNumber: FixedPointNumber + One + Zero + Debug + TryFrom<FixedU128>,
+	{
+		// Forcefully set chain status to running when benchmarking so that the oracle doesn't fail
+		Security::set_status(StatusCode::Running);
+
+		let key = OracleKey::ExchangeRate(currency_id);
+		let rate = FixedU128::checked_from_rational(100, 1).expect("This is a valid ratio");
+		// Needed for feeding values to the oracle but is never used in the oracle
+		let account = AccountId::from([0u8; 32]);
+		Oracle::feed_values(account, vec![(key.clone(), rate)]);
```

Comment:
Every time `get_price` gets called we are also feeding the values, so in the benchmarks we would be measuring this operation which won't happen in the non-benchmark case. 

I don't know how much the detail of the benchmark really matters, but perhaps we could have a function `feed_price` that is only used with `runtime-benchmarks` instead of two `get_prices`, and in the benchmarks code we can just call `feed_price` before the call to the extrinsic. WDYT?

Reply:
This is a good point 👍🏼 However, I think that if we do it this way, benchmarks will become tightly coupled to the Oracle pallet and I'm not so sure how big the differences will be benchmarking wise. Is it worth doing this? Maybe @ebma has a different opinion

Reply:
That's indeed a nice catch @gianfra-t 👍 
We certainly have some inaccuracy here, also considering that the call to `Security::set_status` will also result in an extra write operation. I agree with @bogdanS98 that it might be too much of a hassle to make this perfect because we'd have to introduce some coupling and the structure becomes more complex overall. 

However, we could at least think about adding an extra check at the beginning of this function, which checks if the `Oracle` already has a price for the requested `currency_id`. Then we can at least minimize the overhead of this function in case it gets called multiple times.

Reply:
That makes more sense. With that extra check only one of the 50(ish) samples would get benchmarked.

Reply:
Added the check and regenerated the weights 👍🏼